### ISSUE TITLE
Update of laravel-acl killed my application

### ISSUE THREAD

**Issue by productiveparks** [kodeine/laravel-acl#115]
I have enjoyed using laravel-acl in my program since last summer. But...
I ran composer update to install a new unrelated package and it apparently updated laravel-acl. I have this in my composer.json:
"kodeine/laravel-acl": "~1.0@dev",

Composer pulled down a newer version with a call to BladeCompiler::directive which apparently isn't supported in 5.0. 
So, now my app won't start and I can't run composer any more. I'm basically dead in the water.

What can I do to get back up and running with the older version I was using?

**Comment by kodeine** [comment]
@productiveparks
Can you run update again and try? There had been recent changes.

**Comment by productiveparks** [comment]
The update I ran yesterday is what caused the trouble.
I did a git revert to get my local branch back to where it was before I ran update so I'm OK, but I am concerned because it looks like I won't be able to update laravel-acl unless I upgrade to a more recent version of Laravel.

What is the story with BladeCompiler::directive?
It appears to not be supported in the 5.0 version of Laravel I am running but it is in the 1.0 dev version of laravel-acl.

**Comment by kodeine** [comment]
can you try this function on laravel 5.0

```php
protected function registerBladeExtensions()
    {
        $blade = $this->app['view']->getEngineResolver()->resolve('blade')->getCompiler();
        $blade->extend(function ($view, $compiler) {
            $pattern = $compiler->createMatcher('role');
            return preg_replace($pattern, '<?php if (Auth::check() && Auth::user()->is$2): ?> ', $view);
        });

        $blade->extend(function ($view, $compiler) {
            $pattern = $compiler->createPlainMatcher('endrole');
            return preg_replace($pattern, '<?php endif; ?>', $view);
        });

        $blade->extend(function ($view, $compiler) {
            $pattern = $compiler->createMatcher('permission');
            return preg_replace($pattern, '<?php if (Auth::check() && Auth::user()->can$2): ?> ', $view);
        });

        $blade->extend(function ($view, $compiler) {
            $pattern = $compiler->createPlainMatcher('endpermission');
            return preg_replace($pattern, '<?php endif; ?>', $view);
        });
    }
```

**Comment by kodeine** [comment]
@productiveparks 
please run composer update, i've pushed changes to fix blade template compatibility.

**Comment by productiveparks** [comment]
Thank you!

On Fri, Feb 5, 2016 at 7:52 PM, Kodeine <notifications@github.com> wrote:

> Closed #115 <https://github.com/kodeine/laravel-acl/issues/115>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kodeine/laravel-acl/issues/115#event-541516068>.
>



-- 
Michael Kent Smith
http://www.michaelkentsmith.com
